src/python/pants/engine/internals/build_files_test.py

```diff
@@ -500,6 +520,35 @@ def test_target_adaptor_defaults_applied(target_adaptor_rule_runner: RuleRunner)
     assert target_adaptor.kwargs["tags"] == ["24"]
 
 
+def test_generated_target_defaults(target_adaptor_rule_runner: RuleRunner) -> None:
```

Comment:
This is implemented in `graph.py`, but the `__defaults__` feature, as used in the BUILD files, are otherwise tested here so I felt this was the logical place for this test but could be convinced to move it to `graph_test.py`.